cmd/vulcan-exposed-http-endpoint/main.go

```diff
@@ -45,54 +45,73 @@ If no paths are specified the check will test all the paths present in _paths/*.
 
 // Options defines the options of the check.
 type Options struct {
-	Paths path.Paths `json:"paths"`
+	Paths []path.Path `json:"paths"`
 }
 
 var (
 	checkName = "vulcan-exposed-http-endpoint"
-	// exposedVuln defines the vulnerability that will be returned by the check
-	// when one or more paths are exposed.
+	// exposedVuln defines the vulnerability template that will be returned by
+	// the check when one path is exposed.
 	exposedVuln = report.Vulnerability{
-		Summary:         "Exposed URLs",
-		Description:     "Paths that should not be publicly accessible are exposed.",
+		Summary:         "Exposed URL",
+		Description:     "Path that should not be publicly accessible.",
 		ImpactDetails:   `An attacker may be able to interact with functionalities that could harm your system, like admin endpoints or access information that shouldn't be public.`,
 		Score:           report.SeverityThresholdHigh,
-		Recommendations: []string{"Forbid access to the reported paths."},
-		Resources: []report.ResourcesGroup{
-			report.ResourcesGroup{
-				Name:   "Exposed URLs",
-				Header: []string{"URL"},
-				Rows:   []map[string]string{},
-			},
-		},
+		Recommendations: []string{"Forbid access to the reported path."},
+		Labels:          []string{"issue", "http"},
 	}
 )
 
 func init() {
-	// We don't want to verify the certificates in this checks.
+	// We don't want to verify the certificates in this check.
 	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 }
 
-func exposedURL(l *logrus.Entry, addr *url.URL, paths path.Paths) (*report.Vulnerability, error) {
-	resources := []map[string]string{}
-	var vuln *report.Vulnerability
-	for _, p := range paths {
-		r, err := checkPath(l, addr, p)
+func exposedURL(l *logrus.Entry, addr *url.URL, paths path.Paths) (*[]report.Vulnerability, error) {
+	var vulnerablePaths = map[string][]path.Path{}
+	for p := range paths {
+		url, err := checkPath(l, addr, p)
 		if err != nil {
 			return nil, err
 		}
-		if r != nil {
-			resources = append(resources, r)
+		if url == "" {
+			continue
 		}
+		matches, ok := vulnerablePaths[url]
+		if !ok {
+			matches = make([]path.Path, 1)
+		}
+		matches = append(matches, p)
+		vulnerablePaths[url] = matches
+	}
+	if len(vulnerablePaths) == 0 {
+		return nil, nil
 	}
-	if len(resources) > 0 {
-		vuln = &exposedVuln
-		vuln.Resources[0].Rows = resources
+	var vulns []report.Vulnerability = make([]report.Vulnerability, 0, len(vulnerablePaths))
+	for url, paths := range vulnerablePaths {
+		v := exposedVuln
+		v.AffectedResource = url
+		v.Fingerprint = fingerprint(paths)
```

Comment:
I think that the fingerprint should include `the context` in which the vulnerability has been found.
For example, in this check I would recommend to use only options `Status` and `RegExp`.
We have the following hierarchy:
Target -> AffectedResource -> Fingerprint
Making use of the AffectedResource as Fingerprint in this case would be redundant and won't make us able to detect if the finding conditions (status or regexp) has changed in order to detect the vulnerability.

Reply:
As we discussed, I just deprecated the check.